dubbo-common/src/main/java/org/apache/dubbo/common/store/support/SimpleDataStore.java

```diff
@@ -60,5 +68,22 @@ public void remove(String componentName, String key) {
             return;
         }
         data.get(componentName).remove(key);
+        notifyListeners(componentName, key, null);
+    }
+
+    @Override
+    public void addListener(DataStoreUpdateListener dataStoreUpdateListener) {
+        listeners.add(dataStoreUpdateListener);
+    }
+
+    private void notifyListeners(String componentName, String key, Object value) {
+        for (DataStoreUpdateListener listener : listeners) {
+            try {
+                listener.onUpdate(componentName, key, value);
+            } catch (Throwable t) {
+                logger.warn(
```

Comment:
Can the log print componentName, key, and value variables at the same time to facilitate troubleshooting?